diff --git a/src/network/requestQueue/index.js b/src/network/requestQueue/index.js
index 507a9e2..13d4a6c 100644
--- a/src/network/requestQueue/index.js
+++ b/src/network/requestQueue/index.js
@@ -306,14 +306,16 @@ module.exports = class RequestQueue extends EventEmitter {
   scheduleCheckPendingRequests() {
     // If we're throttled: Schedule checkPendingRequests when the throttle
     // should be resolved. If there is already something scheduled we assume that that
     // will be fine, and potentially fix up a new timeout if needed at that time.
     // Note that if we're merely "overloaded" by having too many inflight requests
     // we will anyways check the queue when one of them gets fulfilled.
+    if (this.pending.length === 0) {
+      return
+    }
+
     let scheduleAt = this.throttledUntil - Date.now()
     if (!this.throttleCheckTimeoutId) {
-      if (this.pending.length > 0) {
-        scheduleAt = scheduleAt > 0 ? scheduleAt : CHECK_PENDING_REQUESTS_INTERVAL
-      }
+      scheduleAt = scheduleAt > 0 ? scheduleAt : CHECK_PENDING_REQUESTS_INTERVAL
       this.throttleCheckTimeoutId = setTimeout(() => {
         this.throttleCheckTimeoutId = null
         this.checkPendingRequests()
